src/renderers/shared/reconciler/instantiateReactComponent.js

```diff
@@ -83,6 +106,8 @@ function instantiateReactComponent(node, parentCompositeType) {
       // represenations. I.e. ART. Once those are updated to use the string
       // representation, we can drop this code path.
       instance = new element.type(element);
+    } else if (isStatelessComponentType(element.type)) {
```

Comment:
It was explicitly added in 0.13 because we don't want to be opinionated about how to structure classes. That is especially important for integration with compile-to-js languages.